libcontainer/container_linux.go

```diff
@@ -112,28 +112,33 @@ func (c *Container) OCIState() (*specs.State, error) {
 	return c.currentOCIState()
 }
 
+// ignoreCgroupError filters out cgroup-related errors that can be ignored,
+// because the container is stopped and its cgroup is gone.
+func (c *Container) ignoreCgroupError(err error) error {
+	if err == nil {
+		return nil
+	}
+	if !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	if c.runType() == Stopped && !c.cgroupManager.Exists() {
+		return nil
+	}
+	return err
+}
+
 // Processes returns the PIDs inside this container. The PIDs are in the
 // namespace of the calling process.
 //
 // Some of the returned PIDs may no longer refer to processes in the container,
 // unless the container state is PAUSED in which case every PID in the slice is
 // valid.
 func (c *Container) Processes() ([]int, error) {
-	var pids []int
-	status, err := c.currentStatus()
-	if err != nil {
-		return pids, err
-	}
-	// for systemd cgroup, the unit's cgroup path will be auto removed if container's all processes exited
-	if status == Stopped && !c.cgroupManager.Exists() {
```

Comment:
In case relevant, and that code also needs a similar fix; note that there's some _very similar_ logic in `container.Signal()`; https://github.com/opencontainers/runc/blob/941e5924efdd5b42c06e1915824f674a3e831bfe/libcontainer/container_linux.go#L358-L369

Reply:
Check the commit description, I explain why I'm doing it the way I'm doing in there.

Reply:
❤️ sorry, should've done so (was looking in between other things 🥹🙈)

Reply:
Added a note to the code why we do the same/similar check both before and after.